add README (#2)
viam-labs/berry-imu: README.md

> # Berry-IMU

Reviewer: Let's add the meta json too in this pr.

Author: That gets generated when you upload the module right? will add

This module provides an implementation of the [BerryIMUv3](https://ozzmaker.com/product/berryimu-accelerometer-gyroscope-magnetometer-barometricaltitude-sensor/) using an i2c connection on ARM64 systems.

## Setup

###  Wiring
Ensure i2c communication is enabled on your single board computer. Wire the SCL and SDA pins on the board to the corresponding pins on the IMU. 
The IMU can be powered with the 3.3v pin. You can verify the i2c connection by running `i2cdetect -y 1` on your board. If the IMU is wired correctly, you will see 
the i2c addresses `1c` and `6a`. 

### Registry
This module can be installed via the viam registry. 

- Go to your robot's page on app.viam.com.
- Click on the *Create Component* button in the Components section.
- Search for the *berryimu* module and select it. 

> This will automatically install the module to your robot.

Author: Should I also have instructions on how to run locally?

Reviewer: No it's okay - maybe point to the docs for running a module locally

IMU component config in the viam app.

# Attributes and Sample Config 
The attributes for the module are as follows:
   - `i2c_bus` (required): the name of the i2c bus wired to the IMU.
   - `calibrate`: set to true to calibrate the magnetometer using the min and max values.
   - `hard_iron_x_max`, `hard_iron_y_max`, `hard_iron_x_min`, `hard_iron_y_min`, `soft_iron_x_max`, `soft_iron_y_max`, `soft_iron_x_min`, `soft_iron_y_min`, (required when calibrate is `true`): values obtained from
    calibration script.

```
    {
      "components": [
    {
      "name": "myberryimu"
      "attributes": {
        "hard_iron_y_max": 1953,
        "soft_iron_x_max": 1069,
        "hard_iron_x_min": -1953,
        "hard_iron_x_max": 118,
        "soft_iron_x_min": -1069,
        "soft_iron_y_max": 1069,
        "soft_iron_y_min": -1056,
        "hard_iron_y_min": -1029
        "calibrate": true,
        "i2c_bus": "1",
      },
      "namespace": "viam-labs"
      "type": "movement_sensor"
      "model": "viam-labs:movement_sernsor:berryimu"
            }
        ]
    }

> ```

Reviewer: Is the the most updated way to configure modules? I was expecting a config that adds the modules and the components from that module into the config field

Author: this is correct for the components section, ill also add the modules section to the sample to reduce confusion

# Berry-IMU

> This module provides an implementation of the [BerryIMUv3](https://ozzmaker.com/product/berryimu-accelerometer-gyroscope-magnetometer-barometricaltitude-sensor/) using an i2c connection on ARM64 systems.

Reviewer 2: Sorry, one last comment! Let's expand this with a direct link to the module's registry page as well:

```suggestion
This module provides an implementation of the [BerryIMUv3](https://ozzmaker.com/product/berryimu-accelerometer-gyroscope-magnetometer-barometricaltitude-sensor/) using an i2c connection on ARM64 systems, and is [available from the Viam registry](https://app.viam.com/module/viam-labs/berryimu).
```

## Setup

###  Wiring
Ensure i2c communication is enabled on your single board computer. Wire the SCL and SDA pins on the board to the corresponding pins on the IMU. 
The IMU can be powered with the 3.3v pin. You can verify the i2c connection by running `i2cdetect -y 1` on your board. If the IMU is wired correctly, you will see 

> the i2c addresses `1c` and `6a`. 

Reviewer 2: If desired, you could also link to the docs instructions for adding I2C support to a RPi. Maybe something like:

```suggestion
- Ensure i2c communication is enabled on your single board computer. For example, if you are using a Raspberry Pi, see [Enable communications protocols](https://docs.viam.com/installation/prepare/rpi-setup/#enable-communication-protocols).
- Wire the SCL and SDA pins on the board to the corresponding pins on the IMU. The IMU can be powered with the 3.3v pin.
- You can verify the i2c connection by running `i2cdetect -y 1` on your board. If the IMU is wired correctly, you will see the i2c addresses `1c` and `6a`. 
```

If you don't want to focus so much on the RPi usage case, feel free to disregard!

### Registry
This module can be installed via the viam registry. 

- Go to your robot's page on app.viam.com.
- Click on the *Create Component* button in the Components section.
- Search for the *berryimu* module and select it. 

This will automatically install the module to your robot.

> You can also [run the module locally](https://docs.viam.com/registry/configure/#local-modules).

Reviewer 2: Nit: I would rely on the docs instructions, which also covers local modules on the same page. What do you think about this version, similar to the [agilex LIMO README](https://github.com/viam-labs/agilex):

```suggestion
### Add the `berryimu` module

To use this module, follow the instructions to [add a module from the Viam Registry](https://docs.viam.com/registry/configure/#add-a-modular-resource-from-the-viam-registry), and search for the `berryimu` movement sensor.
Click **Add module**, choose a name, and click **Create** to add the `berryimu` module and the modular movement sensor it provides.

You can also [run the module locally](https://docs.viam.com/registry/configure/#local-modules).
```

# Calibration
The BerryIMU's magnetometer needs to be calibrated prior to using this module. Running the calibration script in 'src/calibration.py'
will provide configuration values to correct for soft and hard iron distortions. 

be sure to have the viam python sdk installed on your machine before running the calibration script:
```
pip install viam-sdk

> ```

Reviewer 2: ````suggestion
To run the calibration script, you must install the Viam Python SDK on your single board computer:

```sh
pip install viam-sdk
```
````

> Run the calibration script on your single board computer by calling `python3 calibration.py` in the `src` folder.

Reviewer 2: ````suggestion
Then, navigate to the `src` directory in the Python SDK, and run the calibration script using the following command:

```sh
python3 calibration.py
```
````

Bumped SBC mention up to "install Py SDK" step.

Reviewer 2: Ah, sorry, you're right: `src` for this module, not the Python SDK. Thanks for catching! 👍

Author: no worries!!

The script will prompt you to rotate the IMU in all directions. Make sure you are rotating the IMU 360 degrees in the X, Y, and Z directions 
to ensure an accurate calibration. Keep rotating the IMU until the script outputs the calibration values. Copy and paste these values into your

> IMU component config in the viam app.

Reviewer 2: ```suggestion
IMU component config in the [Viam app](https://app.viam.com/robots).
```

# Attributes and Sample Config 
The attributes for the module are as follows:

>    - `i2c_bus` (required): the name of the i2c bus wired to the IMU.
>    - `calibrate`: set to true to calibrate the magnetometer using the min and max values.
>    - `hard_iron_x_max`, `hard_iron_y_max`, `hard_iron_x_min`, `hard_iron_y_min`, `soft_iron_x_max`, `soft_iron_y_max`, `soft_iron_x_min`, `soft_iron_y_min`, (required when calibrate is `true`): values obtained from
>     calibration script.

Reviewer 2: ```suggestion
   - `i2c_bus`: (**Required**) The name of the i2c bus wired to the IMU.
   - `calibrate`: Set to true to calibrate the magnetometer using the min and max values.
   - `hard_iron_x_max`, `hard_iron_y_max`, `hard_iron_x_min`, `hard_iron_y_min`, `soft_iron_x_max`, `soft_iron_y_max`, `soft_iron_x_min`, `soft_iron_y_min`: (**Required** when calibrate is `true`) The calibration values obtained from running the `calibration.py` script.
```

```
    {
    "modules": [
    {
      "type": "registry",
      "name": "berryimu",
      "module_id": "viam-labs:berryimu",
      "version": "0.0.2"
    }
    ],
      "components": [
    {

>       "name": "myberryimu"
>       "namespace": "rdk"
>       "type": "movement_sensor"
>       "model": "viam-labs:movement_sensor:berryimu"
>       "attributes": {
>         "hard_iron_y_max": 1953,
>         "soft_iron_x_max": 1069,
>         "hard_iron_x_min": -1953,
>         "hard_iron_x_max": 118,
>         "soft_iron_x_min": -1069,
>         "soft_iron_y_max": 1069,
>         "soft_iron_y_min": -1056,
>         "hard_iron_y_min": -1029
>         "calibrate": true,
>         "i2c_bus": "1",

Reviewer 2: ```suggestion
      "name": "myberryimu",
      "namespace": "rdk",
      "type": "movement_sensor",
      "model": "viam-labs:movement_sensor:berryimu",
      "attributes": {
        "hard_iron_y_max": 1953,
        "soft_iron_x_max": 1069,
        "hard_iron_x_min": -1953,
        "hard_iron_x_max": 118,
        "soft_iron_x_min": -1069,
        "soft_iron_y_max": 1069,
        "soft_iron_y_min": -1056,
        "hard_iron_y_min": -1029,
        "calibrate": true,
        "i2c_bus": "1"
```
